plugins/parsers/hep/parser.go

```diff
@@ -0,0 +1,226 @@
+package hep
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+var headerNames = map[string]int{"version": 1,
+	"protocol":  2,
+	"src_ip4":   3,
+	"dest_ip4":  4,
+	"src_ip6":   5,
+	"dest_ip6":  6,
+	"src_port":  7,
+	"dest_port": 8,
+	"tsec":      9,
+	//"tmsec":      10,
+	"proto_type": 11,
+	"node_id":    12,
+	"mode_pw":    14,
+	"cid":        17,
+	"vlan":       18,
+	"nodename":   19,
+}
+
+var headerReverseMap = map[int]string{1: "version",
+	2: "protocol",
+	3: "src_ip4",
+	4: "dest_ip4",
+	5: "src_ip6",
+	6: "dest_ip6",
+	7: "src_port",
+	8: "dest_port",
+	9: "tsec",
+	//10: "tmsec",
+	11: "proto_type",
+	12: "node_id",
+	14: "node_pw",
+	17: "cid",
+	18: "vlan",
+	19: "nodename",
+}
+
+type Parser struct {
+	//Json fields
+	MetricName         string
+	HepMeasurementName string
+	TagKeys            []string
+	JSONNameKey        string
+	JSONStringFields   []string
+	JSONQuery          string
+	JSONTimeKey        string
+	JSONTimeFormat     string
+	JSONTimezone       string
+	DefaultTags        map[string]string
+	HepHeader          []string
+}
+
+// DecodeHEP returns a parsed HEP message
+func DecodeHEP(packet []byte) (*HEP, error) {
+	hep := &HEP{}
+	err := hep.parse(packet)
+	if err != nil {
+		return nil, err
+	}
+	return hep, nil
+}
+
+func (h *HEP) parse(packet []byte) error {
+	var err error
+	err = h.parseHEP(packet)
+	if err != nil {
+		return err
+	}
+	t := time.Now()
+	if h.ProtoType == 0 {
+		return nil
+	}
+
+	h.Timestamp = time.Unix(int64(h.Tsec), int64(h.Tmsec*1000))
+	d := t.Sub(h.Timestamp)
+	if d < 0 || (h.Tsec == 0 && h.Tmsec == 0) {
+		h.Timestamp = t
+	}
+
+	if h.NodeName == "" {
+		h.NodeName = strconv.FormatUint(uint64(h.NodeID), 10)
+	}
+	return nil
+}
+
+func (h *Parser) Parse(packet []byte) ([]telegraf.Metric, error) {
+
+	hep, err := DecodeHEP(packet)
+
+	if err != nil {
+		return nil, err
+	}
+	if len(h.HepMeasurementName) != 0 {
+		h.MetricName = h.HepMeasurementName
+	} else {
+		h.MetricName = "hep"
+	}
+	headerTags := make(map[string]string)
+	jsonParser, err := json.New(
```

Comment:
Please create the JSON-parser only once per `Parser` instead of doing it in each call to `Parse()`.